package haptix.comm.msgs;

/// \ingroup haptix_comm_msgs
/// \interface hxRobot
/// \brief Robot information.
/// This message specifies inherent properties of the robot that do not change
/// during simulation (for example, the number of joints in the robot arm).

message hxRobot
{
  message hxLimit
  {
    required float minimum = 1;
    required float maximum = 2;
  }

  required int32 motor_count          = 1;
  required int32 joint_count          = 2;
  required int32 contact_sensor_count = 3;
  required int32 imu_count            = 4;
  repeated hxLimit joint_limit        = 5;
  repeated hxLimit motor_limit        = 6;
  required float update_rate          = 7;
}
